src/cider_nrepl/main.clj

```diff
@@ -28,17 +28,36 @@
   [middleware-var-strs]
   (into [] mw-xf middleware-var-strs))
 
+(defn- build-handler
+  [{:keys [handler middleware]}]
+  (let [handler (or handler nrepl.server/default-handler)]
+    (if middleware
+      (apply handler (->mw-list middleware))
+      (handler))))
+
 (defn start-nrepl
-  [opts]
-  (let [{:keys [handler middleware bind port] :or {port 0 bind "::"}} opts
+  "Starts a socket-based nREPL server. Accepts a map with the following keys:
+ 
+   * :port — defaults to 0, which autoselects an open port
+
+   * :bind — bind address, by default \"::\" (falling back to \"localhost\" if
+     \"::\" isn't resolved by the underlying network stack)
+
+   * :handler — the nREPL message handler to use for each incoming connection;
+     defaults to the result of `(nrepl.server/default-handler)`
 
-        handler (cond-> (or handler nrepl.server/default-handler)
-                  middleware (apply (->mw-list middleware)))
+   * :middleware - a sequence of vars or string which can be resolved to vars,
+     representing middleware you wish to mix in to the nREPL handler. Vars can
+     resolve to a sequence of vars, in which case they'll be flattened into the
+     list of middleware."
+  [{:keys [bind port] :as opts}]
+  (let [handler
+        (build-handler opts)
```

Comment:
Here I'd `(or handler (build-handler)`. (provided it's extracted from the `ops` map above.